src/app/address/controllers/addressResults.js

```diff
@@ -8,16 +12,33 @@ class AddressResultsController extends BaseController {
     });
   }
 
+  validateFields(req, res, callback) {
+    const formFields = req.form.options.fields;
+    const addresses = req.sessionModel.get("addresses");
+
+    if (addresses) {
+      const currentAddress = addresses[0];
+
+      const selectedAddress = req.form.values.addressResults;
+      const searchResults = req.sessionModel.get("searchResults");
+      const chosenAddress = this.getAddress(selectedAddress, searchResults);
+
+      formFields.addressResults?.validate.push({
+        fn: addressSelectorValidator,
+        arguments: [chosenAddress, currentAddress],
+      });
+    }
+
+    super.validateFields(req, res, callback);
+  }
+
   saveValues(req, res, callback) {
     super.saveValues(req, res, () => {
       try {
         const selectedAddress = req.form.values.addressResults;
-        const allAddresses = req.sessionModel.get("searchResults");
+        const searchResults = req.sessionModel.get("searchResults");
 
-        const chosenAddress = Object.assign(
-          {},
-          allAddresses.find((address) => address.text === selectedAddress)
-        ); // force deep copy
+        const chosenAddress = this.getAddress(selectedAddress, searchResults);
```

Comment:
extract out the logic for getting the chosen address. So it can be reused in validation.